### PR TITLE
Migrate contracts job for Celo

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Added a `migrate-and-publish-celo` job which performs contract migration on Celo chain. For now it publishes artifacts to GCS bucket only. Publishing to Tenderly and NPM should be added later.

Apart from that, several changes have been done to implement support of the Celo chain:
- Added new `alfajores` network in Truffle config
- Added `alfajores` support in migration scripts
- Refactored the former `circleci-provision-external-contracts.sh` script in order to be simpler and work with multiple chains
- Updated the relay genesis config in order to allow the maintainer to catch-up testnet BTC head quicker

### CODE REVIEW

File: .github/workflows/contracts.yml [hunk 1]
```diff
@@ -81,4 +82,69 @@ jobs:
...
+      - uses: google-github-actions/setup-gcloud@v0.2.0
+        with:
+          project_id: ${{ secrets.GOOGLE_PROJECT_ID }}
+          service_account_key: ${{ secrets.KEEP_TEST_GCR_JSON_KEY }}
+
+      - name: Fetch external contracts artifacts
+        env:
+          CONTRACT_DATA_BUCKET: ${{ secrets.KEEP_TEST_CONTRACT_DATA_BUCKET }}
+          CONTRACT_DATA_BUCKET_DIR: keep-ecdsa-celo
+          ETH_NETWORK_ID: 44787 # Alfajores network ID
```
Comment (michalinacienciala) [inline .github/workflows/contracts.yml:132]:
```suggestion
          ETH_NETWORK_ID: ${{ secrets.KEEP_TEST_CELO_NETWORK_ID }}
```

File: .github/workflows/contracts.yml [hunk 2]
```diff
@@ -81,4 +82,69 @@ jobs:
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./solidity
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12.x"
```
Comment (michalinacienciala) [inline .github/workflows/contracts.yml:106]: We could switch to using node matrix here - but this is something that I can change when working on migration job job for Ethereum chain, so you can leave it as it is...

File: .github/workflows/contracts.yml [hunk 3]
```diff
@@ -81,4 +82,69 @@ jobs:
...
+
+  migrate-and-publish-celo:
+    needs: [build-and-test, lint]
+    # FIXME: alfajores branch and pull request events are for test purposes only - revert once done
+    if: |
+      (startsWith(github.head_ref, 'releases/alfajores/')
+        || github.ref == 'refs/heads/master')
+        && (github.event_name == 'push'
+        || github.event_name == 'pull_request'
+        || github.event_name == 'workflow_dispatch')
```
Comment (michalinacienciala) [inline .github/workflows/contracts.yml:95]: Can the final config be applied now? Or do we want to remove it in different PR?

File: .github/workflows/contracts.yml [hunk 4]
```diff
@@ -81,4 +82,69 @@ jobs:
...
+          CONTRACT_DATA_BUCKET: ${{ secrets.KEEP_TEST_CONTRACT_DATA_BUCKET }}
+          CONTRACT_DATA_BUCKET_DIR: keep-ecdsa-celo
+          ETH_NETWORK_ID: 44787 # Alfajores network ID
+        run: ./scripts/ci-provision-external-contracts.sh
+
+      - name: Migrate contracts
+        env:
+          CONTRACT_OWNER_CELO_ACCOUNT_PRIVATE_KEY: |
+            ${{ secrets.KEEP_TEST_CELO_CONTRACT_OWNER_PRIVATE_KEY }}
+        run: npx truffle migrate --reset --network alfajores
```
Comment (michalinacienciala) [inline .github/workflows/contracts.yml:139]:
```suggestion
        env:
          TRUFFLE_NETWORK: ${{ secrets.KEEP_TEST_CELO_TRUFFLE_NETWORK }}
          CONTRACT_OWNER_CELO_ACCOUNT_PRIVATE_KEY: |
            ${{ secrets.KEEP_TEST_CELO_CONTRACT_OWNER_PRIVATE_KEY }}
        run: npx truffle migrate --reset --network $TRUFFLE_NETWORK
```

File: .github/workflows/contracts.yml [hunk 5]
```diff
@@ -81,4 +82,69 @@ jobs:
...
+
+      # TODO: if Celo works with Tenderly, add appropriate push step
+
```
Comment (michalinacienciala) [inline .github/workflows/contracts.yml:142]: Comment can be deleted

File: solidity/scripts/ci-provision-external-contracts.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -ex
+
+# Fetch addresses of contacts migrated from keep-network/keep-core project.
```
Comment (michalinacienciala) [inline solidity/scripts/ci-provision-external-contracts.sh:4]:
```suggestion
# Fetch addresses of contacts migrated from keep-network/keep-ecdsa project.
```

**Review by michalinacienciala — COMMENTED** [review]
I reviewed the parts that I understand (workflow-related changes).
Leaving for @nkuba to review as well.

File: .github/workflows/contracts.yml (hunk 1 above)
Comment (lukasz-zimnoch) [reply .github/workflows/contracts.yml]: Please see https://github.com/keep-network/tbtc/commit/41be3be7830fef652f28e27835ebc0803443fb21

File: .github/workflows/contracts.yml (hunk 3 above)
Comment (lukasz-zimnoch) [reply .github/workflows/contracts.yml]: Please see https://github.com/keep-network/tbtc/commit/41be3be7830fef652f28e27835ebc0803443fb21

File: .github/workflows/contracts.yml (hunk 4 above)
Comment (lukasz-zimnoch) [reply .github/workflows/contracts.yml]: Please see https://github.com/keep-network/tbtc/commit/41be3be7830fef652f28e27835ebc0803443fb21

File: .github/workflows/contracts.yml (hunk 5 above)
Comment (lukasz-zimnoch) [reply .github/workflows/contracts.yml]: Please see https://github.com/keep-network/tbtc/commit/41be3be7830fef652f28e27835ebc0803443fb21

File: solidity/scripts/ci-provision-external-contracts.sh (hunk 6 above)
Comment (lukasz-zimnoch) [reply solidity/scripts/ci-provision-external-contracts.sh]: Please see https://github.com/keep-network/tbtc/commit/41be3be7830fef652f28e27835ebc0803443fb21

File: .github/workflows/contracts.yml (hunk 1 above)
Comment (lukasz-zimnoch) [reply .github/workflows/contracts.yml]: Small update: https://github.com/keep-network/tbtc/commit/0f11706235724e250b1707d16aa98cc8cf0731c7

**Review by nkuba — APPROVED** [review]

**Pull request merged**